core/dbt/task/run.py

```diff
@@ -245,6 +251,34 @@ def __init__(self, args, config):
         super().__init__(args, config)
         self.ran_hooks = []
         self._total_executed = 0
+        self.deferred_manifest: Optional[WritableManifest] = None
+
+    def _get_state_path(self) -> Path:
+        if self.args.state is not None:
+            return self.args.state
+        elif ARTIFACT_STATE_PATH is not None:
```

Comment:
Nice! Glad to see that `ARTIFACT_STATE_PATH` and `--state` are mutually substitutable, to support Cloud deployment.

I _think_ the conclusion of [this conversation](https://github.com/fishtown-analytics/dbt-cloud/issues/1497#issuecomment-663203559) was to add a separate environment variable for deferral as well. Let me know if you disagree.

Reply:
I did that! It's set up differently because it's a boolean rather than a value, but it's defined in `dbt/flags.py`. It being set/unset informs the default value for the `--defer`/`--no-defer` argument to change.